web-ui/src/main/resources/catalog/js/edit/DirectoryController.js

```diff
@@ -323,34 +330,149 @@
                   timeout: 0,
                   type: 'danger'});
               } else {
-                searchEntries();
-                $scope.isImporting = false;
-                $scope.xml = null;
+                refreshEntriesInfo();
+                $scope.closeEditor();
               }
             }
         );
       };
 
-      $scope.startSharing = function() {
-        $('#gn-share').modal('show');
-      };
-
-      $scope.$on('PrivilegesUpdated', function() {
-        $('#gn-share').modal('hide');
-      });
+      // ACTIONS
 
-      $scope.delEntry = function(e) {
+      $scope.delEntry = function (e) {
         // md.delete?uuid=b09b1b16-769f-4dad-b213-fc25cfa9adc7
-        gnMetadataManager.remove(e['geonet:info'].id).then(searchEntries);
+        gnMetadataManager.remove(e['geonet:info'].id).then(refreshEntriesInfo);
       };
 
-      $scope.copyEntry = function(e) {
+      $scope.copyEntry = function (e) {
         //md.create?id=181&group=2&isTemplate=s&currTab=simple
         gnMetadataManager.copy(e['geonet:info'].id, $scope.ownerGroup,
-            fullPrivileges,
-            'SUB_TEMPLATE').then(searchEntries);
+          fullPrivileges,
+          e.isTemplate === 't' ? 'TEMPLATE_OF_SUB_TEMPLATE' : 'SUB_TEMPLATE'
+        ).then(refreshEntriesInfo);
+      };
+
+      $scope.convertToTemplate = function (e) {
+        if (e.isTemplate !== 's') {
+          $rootScope.$broadcast('StatusUpdated', {
+            title: $translate.instant('notADirectoryEntry'),
+            error: '',
+            timeout: 0,
+            type: 'danger'});
+          return;
+        }
+
+        // conversion to template is done by duplicating into template type
+        // the original entry is kept
+        gnMetadataManager.copy(e['geonet:info'].id, $scope.ownerGroup,
```

Comment:
I can't find of a way to do this post successfully. I'm doing:
`POST api/records/3194900/editor?&template=t&id=3231011&version=1&minor=true`

Problems are:
* I can't find a way to find the `version` of the object to update, which is apparently required by the editor API
* When I get the version right & do the request above (with a valid ID), the request fails on a 404 with the message:
```xml
<apiError>
    <code>
        resource_not_found
    </code>
    <description>
        Requested metadata with id '3231011' is not available in current session. Open an editing session on this record first.
    </description>
    <message>
        ResourceNotFoundException
    </message>
</apiError>
```
Does the API filter entries to automatically exclude subtemplates?